multicluster/controllers/multicluster/label_identity_controller.go

```diff
@@ -0,0 +1,311 @@
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package multicluster
+
+import (
+	"context"
+	"sync"
+
+	v1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/klog/v2"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	mcsv1alpha1 "antrea.io/antrea/multicluster/apis/multicluster/v1alpha1"
+	"antrea.io/antrea/multicluster/controllers/multicluster/common"
+	"antrea.io/antrea/multicluster/controllers/multicluster/commonarea"
+)
+
+// LabelIdentityReconciler watches relevant Pod and Namespace events in the member cluster,
+// computes the list of unique label identities in the cluster, and exports it to the
+// leader cluster for further processing.
+type (
+	LabelIdentityReconciler struct {
+		client.Client
+		Scheme           *runtime.Scheme
+		commonAreaGetter RemoteCommonAreaGetter
+		remoteCommonArea commonarea.RemoteCommonArea
+		// labelMutex prevents concurrent access to labelToPodsCache and podLabelCache
+		labelMutex sync.RWMutex
+		// resourceExportMutex prevents concrrent updates to the cluster label identity ResourceExport
+		resourceExportMutex sync.Mutex
+		// labelToPodsCache stores mapping from label identities to Pods that have this label identity
+		labelToPodsCache map[string]sets.String
+		// podLabelCache stores mapping from Pods to their label identities
+		podLabelCache  map[string]string
+		localClusterID string
+	}
+)
+
+func NewLabelIdentityReconciler(
+	client client.Client,
+	scheme *runtime.Scheme,
+	commonAreaGetter RemoteCommonAreaGetter) *LabelIdentityReconciler {
+	return &LabelIdentityReconciler{
+		Client:           client,
+		Scheme:           scheme,
+		commonAreaGetter: commonAreaGetter,
+		labelToPodsCache: map[string]sets.String{},
+		podLabelCache:    map[string]string{},
+	}
+}
+
+// +kubebuilder:rbac:groups=multicluster.crd.antrea.io,resources=resourceexports,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups="",resources=pods,verbs=get;list;watch
+// +kubebuilder:rbac:groups="",resources=namespaces,verbs=get;list;watch
+func (r *LabelIdentityReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
```

Comment:
Will it be more clear if we check the req.NamespacedName to determine if it's Namespace or Pod first?
Namespace's req.NamespacedName should have empty Namespace.

Reply:
Done.